gobblin-runtime/src/main/java/gobblin/runtime/mapreduce/MRJobLauncher.java

```diff
@@ -594,7 +594,9 @@ protected void cleanup(Context context) throws IOException, InterruptedException
         // Ignored
       } finally {
         if (this.jobMetrics.isPresent()) {
+          this.jobMetrics.get().triggerMetricReporting();
```

Comment:
Stop metric reporting kills the Kafka connections, etc. so the order cannot be switched. I don't think there is an issue with two reports happening at the same time, metrics will be duplicated, but there was never a guarantee otherwise.